packages/react-instantsearch-hooks-dom/src/ui/__tests__/InfiniteHits.test.tsx

```diff
@@ -0,0 +1,181 @@
+    const props = createProps({});
+
+    const { container } = render(<InfiniteHits {...props} />);
+
+    expect(container).toMatchInlineSnapshot(`
+      <div>
+        <div
+          class="ais-InfiniteHits"
+        >
+          <button
+            class="ais-InfiniteHits-loadPrevious ais-InfiniteHits-loadPrevious--disabled"
+            disabled=""
+          >
+            Show previous results
+          </button>
+          <ol
+            class="ais-InfiniteHits-list"
+          >
+            <li
+              class="ais-InfiniteHits-item"
+            >
+              <div
+                style="word-break: break-all;"
+              >
+                {"objectID":"abc","__position":1}
+                …
+              </div>
+            </li>
+            <li
+              class="ais-InfiniteHits-item"
+            >
+              <div
+                style="word-break: break-all;"
+              >
+                {"objectID":"def","__position":2}
+                …
+              </div>
+            </li>
+          </ol>
+          <button
+            class="ais-InfiniteHits-loadMore"
+          >
+            Show more results
+          </button>
+        </div>
+      </div>
+    `);
+  });
+
+  test('renders custom className', () => {
+    const props = createProps({
+      className: 'extra',
+    });
+
+    const { container } = render(<InfiniteHits {...props} />);
+
+    expect(container.querySelector('.ais-InfiniteHits')!.className).toBe(
+      'ais-InfiniteHits extra'
+    );
+  });
+
+  test('renders with custom hitComponent', () => {
+    const props = createProps({
+      hitComponent: ({ hit }) => <strong>{hit.objectID}</strong>,
+    });
+
+    const { container } = render(<InfiniteHits {...props} />);
+
+    expect(container.querySelectorAll('strong')).toHaveLength(2);
+
+    expect(container).toMatchInlineSnapshot(`
+      <div>
+        <div
+          class="ais-InfiniteHits"
+        >
+          <button
+            class="ais-InfiniteHits-loadPrevious ais-InfiniteHits-loadPrevious--disabled"
+            disabled=""
+          >
+            Show previous results
+          </button>
+          <ol
+            class="ais-InfiniteHits-list"
+          >
+            <li
+              class="ais-InfiniteHits-item"
+            >
+              <strong>
+                abc
+              </strong>
+            </li>
+            <li
+              class="ais-InfiniteHits-item"
+            >
+              <strong>
+                def
+              </strong>
+            </li>
+          </ol>
+          <button
+            class="ais-InfiniteHits-loadMore"
+          >
+            Show more results
+          </button>
+        </div>
+      </div>
+    `);
+  });
+
+  test('renders without showPrevious if disabled', () => {
+    const props = createProps({
+      showPrevious: undefined,
+    });
+
+    const { container } = render(<InfiniteHits {...props} />);
+
```

Comment:
Maybe we can ditch the snapshot and have a test like the following, for clarity and readability?

```js
expect(container.querySelector('.ais-InfiniteHits-loadPrevious')).toBeNull();
```